lib/Driver/ToolChains/CommonArgs.cpp

```diff
@@ -870,6 +870,21 @@ tools::ParsePICArgs(const ToolChain &ToolChain, const ArgList &Args) {
     PIP = true;
   }
 
+  // Pagerando and '-ffunction-sections'/'-fdata-sections' are incompatible
```

Comment:
The pros that I can think of: 
This version has the advantage that the Pagerando checks are in one place and not spread out/interleave with other options.  
As you said: if we put it with the other options we don't have to recompute the values.  
Do you want to make a decision on this one? (otherwise I probably leave it as is)